core/test/nuxeo-page-provider.test.js

```diff
@@ -108,4 +108,173 @@ suite('<nuxeo-page-provider>', () => {
       throw new Error('Expected to have an error response!');
     });
   });
+
+  suite('Parameters Transformation', () => {
+    function checkDefaultParameters(params, shouldIncludeKeys = [], expectedNumberOfNamedParameters = 0) {
+      const expectedKeys = ['currentPageIndex', 'namedParameters', 'offset', 'pageSize'];
+      const unexpectedKeys = ['query', 'queryParams', 'quickFilters', 'sortBy', 'sortOrder'];
+
+      shouldIncludeKeys.forEach((key) => {
+        if (unexpectedKeys.includes(key)) {
+          expectedKeys.push(key);
+          unexpectedKeys.splice(key, 1);
+        }
+      });
+
+      expect(params)
+        .to.exist.and.to.be.an('object')
+        .that.has.all.keys(expectedKeys)
+        .and.not.have.keys(unexpectedKeys);
+
+      expect(params.currentPageIndex).to.be.equal(0);
+      expect(Object.keys(params.namedParameters)).to.have.lengthOf(expectedNumberOfNamedParameters);
+      expect(params.offset).to.be.undefined;
+      expect(params.pageSize).to.be.equal(-1);
+    }
+
+    test('Should include query when it is defined and not empty', async () => {
+      const provider = await fixture(html`
+        <nuxeo-page-provider id="nx-pp" query="select * from Document"></nuxeo-page-provider>
+      `);
+
+      const transformedParams = provider._params;
+
+      checkDefaultParameters(transformedParams, ['query']);
+      expect(transformedParams.query).to.be.equal('select * from Document');
+    });
+
+    test('Should not include query when it has an empty string', async () => {
+      const provider = await fixture(html`
+        <nuxeo-page-provider id="nx-pp" query=""></nuxeo-page-provider>
+      `);
+
+      const transformedParams = provider._params;
+      checkDefaultParameters(transformedParams);
```

Comment:
We could just save a line and do:
```suggestion
      checkDefaultParameters(provider._params);
```

Reply:
Of course, thanks!